src/Currencies.java

```diff
@@ -0,0 +1,22 @@
+import java.util.*;
+public class Currencies {
+    Map<String, Map<String,Double>> rates;
+    Currencies(){
+        rates= new HashMap<>();
+    }
+    public void addCurrency (String currency){
+        rates.put(currency,new HashMap<>());
+    }
+    public void addRates(String x, String y, Double i){
```

Comment:
please provide better naming for arguments. It is not easy to guess what those variable are for without reading the whole code (x -> from, y -> to , i -> rate and reading this code would be much simpler)